src/rocdecode/vaapi/vaapi_videodecoder.cpp

```diff
@@ -301,6 +304,42 @@ rocDecStatus VaapiVideoDecoder::SubmitDecode(RocdecPicParams *pPicParams) {
             break;
         }
 
+        case rocDecVideoCodec_AV1: {
+            pPicParams->pic_params.av1.current_frame = curr_surface_id;
+
+            if (pPicParams->pic_params.av1.current_display_picture != 0xFF) {
+                pPicParams->pic_params.av1.current_display_picture = va_surface_ids_[pPicParams->pic_params.av1.current_display_picture];
+            }
+
+            if (pPicParams->pic_params.av1.anchor_frames_num != 0xFF) {
+                for (int i = 0; i < pPicParams->pic_params.av1.anchor_frames_num; i++) {
+                    pPicParams->pic_params.av1.anchor_frames_list[i] = va_surface_ids_[pPicParams->pic_params.av1.anchor_frames_list[i]];
```

Comment:
Add a check of the index against the vector size to prevent invalid memory access.